src/CliCommand/SoxCommand.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace IndieHD\AudioManipulator\CliCommand;
+
+use IndieHD\AudioManipulator\CliCommand\SoxCommandInterface;
+
+class SoxCommand extends CliCommand implements SoxCommandInterface
+{
+    private $name = 'sox_command';
+
+    protected $binary = 'sox';
+
+    protected $parts = [
+        'gopts' => [],      // Global options
+        'fopts-in' => [],   // File options (input)
+        'infile' => [],     // Input file
+        'fopts-out' => [],  // File options (output)
+        'outfile' => [],    // Output file
+        'effect' => [],     // Effect
+        'effopt' => [],     // Effect options
+    ];
+
+    public function name(): string
+    {
+        return $this->name;
+    }
+
+    public function addPart(string $name, string $value): void
```

Comment:
So seeing your approach here I would recommend refactoring the method name from `addPart()` .. I'm not real sure what to name it but you're not really adding the "part" but rather attaching to a part in the parts array ?

Reply:
Good call; you're absolutely correct. What about `addArgumentToPart()`, or even just `addArgument()`?

Reply:
addArgument(); sounds good to me ?